### PR TITLE
💄 Adding origins to historical literacy and expenditure datasets

### PR DESCRIPTION
**Pull request by veronikasamborska1994**
Because the two datasets on expenditure and literacy were fasttracked I am using a custom function in misc to define origins for the historical literacy and expenditure estimates. These are combined with the World Bank EdStats metadata (already updated to a new format) for the subsequent data pages (specifically [this one which is on the list of the most important ones](https://ourworldindata.org/grapher/cross-country-literacy-rates?tab=map)).

### CODE REVIEW

**Comment by veronikasamborska1994** [comment]
hey @pabloarosado I've now moved the metadata reformatting to snapshot and it looks good on grapher!

File: snapshots/education/2018-04-18/literacy_rates.feather.dvc
Comment (pabloarosado) [inline snapshots/education/2018-04-18/literacy_rates.feather.dvc:17]: As explicitly mentioned in [the guidelines](https://docs.owid.io/projects/etl/architecture/metadata/reference/origin/#originproducer), and discussed multiple times, the producer is (almost) never OWID. The only case where this is true is for the `regions` dataset. If you have very strong reasons to choose OWID as a producer, it would be good to discuss it and change the guidelines.
In this case, I suppose the producer could be either "World Bank" or "Various sources".

**Review by pabloarosado — CHANGES_REQUESTED** [review]
Hi @veronikasamborska1994, there are a few things I would change in this PR:
1. It's much better (for reviewers and for future you) to explicitly load all datasets at the beginning of the run function, instead of "secretly" loading them inside `combine_historical_literacy_expenditure`.
2. A minor improvement is to load datasets using: `ds_meadow = paths.load_dataset("education")` (and the same for the other two datasets). This simplifies the code.
3. I don't understand why you use `copy_metadata` multiple times. This is used mostly for operations that do not propagate metadata. But `merge` does propagate origins. When you use `copy_metadata` you are actually **removing** origins. And because of this later you are forced to manually combine origins. Why not letting `merge` propagate origins automatically? So, I'd suggest to remove all instances of `copy_metadata`.
4. The functions `decide_literacy` and `decide_expenditure` look to me like they could be deleted. Instead, you can define `combined_df["combined_literacy"] = combined_df["SE.ADT.LITR.ZS"].fillna(combined_df["literacy_rates__world_bank__cia_world_factbook__and_other_sources"])` (and similar for expenditure), isn't that the intention? Using a fillna would also propagate origins automatically.
5. After you have done the previous changes, I believe you won't need to manually change origins.
6. Be careful when doing `metadata = tb.metadata`, if you change `metadata`, it may alter `tb.metadata` (not totally sure). A safer option would be to do `metadata = tb.metadata.copy()`.
7. I've spotted a `Normalised` somewhere. We should use American spelling, `Normalized`.

I'd suggest to implement those changes, unless I misunderstood something.

**Comment by veronikasamborska1994** [comment]
thanks @pabloarosado! and thanks for spotting the redundancy with .copy_metadata. I think when I first started using the new metadata the pr.merge wasn't working properly and propagating the metadata correctly in every case and this was my solution and looks like I continued just using this as a default. I've made all of the changes you've suggested now. Could you take a look and approve if it all looks ok now (when you get a chance)?

**Comment by veronikasamborska1994** [comment]
@pabloarosado all done now! thanks again for this feedback! also, useful for any future updates I'll have to do (old ---> new metadata format).

File: etl/steps/data/garden/wb/2023-07-10/education.py
Comment (pabloarosado) [inline etl/steps/data/garden/wb/2023-07-10/education.py:85]: Instead of printing, you should assert things. Assert that the columns are identical, then drop one of them. If it changes in the next update, you will not have any way to miss it.

File: etl/steps/data/garden/wb/2023-07-10/education.py
Comment (pabloarosado) [inline etl/steps/data/garden/wb/2023-07-10/education.py:105]: Style: In Python, the first line in a docstring should be imperative, e.g. "Merge historical...".

Comment (pabloarosado) [reply etl/steps/data/garden/wb/2023-07-10/education.py:105]: DataFrame -> Table?

File: snapshots/education/2018-04-18/literacy_rates.feather.dvc
Comment (pabloarosado) [inline snapshots/education/2018-04-18/literacy_rates.feather.dvc:6]: The title should not include the producer.

File: snapshots/education/2018-04-18/literacy_rates.feather.dvc
Comment (pabloarosado) [inline snapshots/education/2018-04-18/literacy_rates.feather.dvc:7]: The sources should be specified in `citation_full` (although there can be some overlap with the description, if it helps describe the dataset).

File: snapshots/education/2018-04-18/literacy_rates.feather.dvc
Comment (pabloarosado) [inline snapshots/education/2018-04-18/literacy_rates.feather.dvc:20]: `url-main` as far as I know should be just one URL (otherwise I can imagine that it will cause issues in grapher). Any additional ones should be in `description` or `citation_full` or elsewhere.

**Review by pabloarosado — APPROVED** [review]
I added some minor style and metadata corrections. After that, feel free to merge!

**Pull request merged**
